cli/module_registry.go

```diff
@@ -637,3 +645,35 @@ func getEntrypointForVersion(mod *apppb.Module, version string) (string, error)
 	// if there is no entrypoint set yet, use the last uploaded entrypoint
 	return mod.Entrypoint, nil
 }
+
+func isTarball(path string) bool {
+	return strings.HasSuffix(path, ".tar.gz") || strings.HasSuffix(path, ".tgz")
+}
+
+func createTarballForUpload(moduleUploadPath string, stdout io.Writer) (string, error) {
+	tmpFile, err := os.CreateTemp("", "module-upload-*.tar.gz")
```

Comment:
We want to create a file instead of using a byte buffer (like board) because a module can be very large (multi GiB).